src/index.js

```diff
@@ -1,53 +1,71 @@
-import { Router } from 'express';
-import Middleware from './Middleware';
+import JwtHelpers from './JwtHelpers';
 import Authenticator from './Authenticator';
+import augmentRequest from './augmentRequest';
 
-/*
-  {
-    id_field: 'id'
-    secret
-    authRules
-  }
- */
-
-export function MakeAuthenticator(config) {
-  return new Authenticator(config);
-}
-
-/*
-  {
-    id_field: 'id'
-    secret
-    authRules
-    fetchUser: // function provided with jwt_payload and done
-    createUser: // function to create user provided with done
-  }
- */
+const derelict = (function(){
+  let useXsrf = true;
+  let authenticator = {};
+
+  return {
+    setup(config) {
+      const jwt = JwtHelpers(config.secret);
+      useXsrf = config.hasOwnProperty(useXsrf) ? config.useXsrf : true;
+      authenticator = Authenticator(jwt, config.createUser, config.fetchUser, useXsrf);
+      augmentRequest(jwt, config.authRules, useXsrf);
+    },
+
+    signUp(req, res) {
+      const newUser = req.body;
+      
+      console.log(newUser);
```

Comment:
You just shouldnt have console output. Unless you want to leave them and then take them out for production. But thats just an extra step later